File: src/School/COP3530/Project3/src/Node.cpp
#include <vector>

using namespace std;

class Node
{
    private:
        string name;
        double score;
        vector<string> genres;
        vector<double> edges;
    public:
        Node(string name, double score, vector<string> genres)
        {
            this->name = name;
            this->score = score;
            this->genres = genres;
        }

        string getName()
        {
            return name;
        }

        double getScore()
        {
            return score;
        }

        vector<string> getGenres()
        {
            return genres;
        }

        vector<double> getEdges()
        {
            return edges;
        }

        void addEdge(double weight)
        {
            edges.push_back(weight);
        }
};

File: src/School/COP3530/P3Python/Node.py
class node:

    def __init__(self, name, score, popularity, genres, summary):
        self.name = name
        self.score = score
        self.popularity = popularity
        self.genres = genres
        self.edges = []
        self.recommendations = []
        self.summary = summary
        self.similarity = 0.0

    def addEdge(self, weight):
        self.edges.append(weight)